src/controller.coffee

```diff
@@ -834,33 +833,31 @@ define ['droplet-helper',
       return clone.end
 
   Editor::spliceOut = (node) ->
-    leading = node.getLeadingText()
-    if node.start.next is node.end.prev
-      trailing = null
-    else
-      trailing = node.getTrailingText()
+    @prepareNode node, null
+    node.spliceOut()
 
-    [leading, trailing] = @mode.parens leading, trailing, node.getReader(), null
+  Editor::spliceIn = (node, location) ->
+    container = location.container ? location.visParent()
+    if container.type is 'block'
+      container = container.visParent()
 
-    node.setLeadingText leading; node.setTrailingText trailing
+    @prepareNode node, container
+    node.spliceIn location
 
-    node.spliceOut()
 
-  Editor::spliceIn = (node, location) ->
+  Editor::prepareNode = (node, context) ->
```

Comment:
Not sure what a better name for this would be. I noticed there was shared logic between spliceIn and spliceOut, and this was logic that I also wanted to call without a context when dropping in ace mode (so that I get the benefit of the call to parens)